carbon-design-system/carbon
# fix(side-nav): update transition-property for fixed nav animation

When using `<SideNav isFixedNav expanded={state}>`, the side nav currently doesn't animate in or out. This PR updates the transition definition accordingly.

#### Changelog

**New**

- Added `transform` to the transitions of `.side-nav`

**Changed**

- Changed `width` to `inline-size` in the transitions of `.side-nav`

#### Testing / Reviewing

1. Pull code locally
2. Add a way to show and hide the sidenav to the "Fixed Side Nav" story (example below)
3. Verify animation is working
4. Verify there are no visual regressions with other side nav variants

---

Example demo story

```jsx
export const FixedSideNav = () => {
  const [expanded, setExpanded] = React.useState(false);

  React.useEffect(() => {
    setInterval(() => {
      setExpanded((s) => {
        setExpanded(!s);
      });
    }, 2000);
  }, []);

  return (
    // [...]
    <SideNav
       // [...]
      expanded={expanded}
    />
    // [...]
  );
}
```
